extras/jslibfetcher.py

```diff
@@ -15,339 +21,386 @@ def process(url, destination):
         print "Failed", e
 
 
-def get_pattern(prefix, url_pattern, versions):
-    for version in versions:
-        url = url_pattern % version
-        process(url, "%s.%s.%s" % (prefix, version, url.split("/")[-1]))
+def get_pattern(prefix, url_pattern, library):
+    for rule, versions in library.iteritems():
+        for version in versions:
+            url = url_pattern % version
+            process(url, rule.lower(),
+                "%s.%s.%s" % (prefix, version, url.split("/")[-1]))
 
 
 process("https://addons.cdn.mozilla.net/en-US/firefox/files/browse/149703/"
         "file-serve/bootstrap.js?token=f885db28-1ec2-419d-8b35-72300523befb",
-        "PersonasInteractive_bootstrap.js")
-
-
-BACKBONE_VERSIONS = [
-    "1.0.0",
-    "1.1.0",
-    "1.1.1",
-    "1.1.2",
-]
-BACKBONE_LOCALSTORAGE_VERSIONS = [
-    "1.1.10",
-    "1.1.12",
-    "1.1.13",
-    "1.1.14",
-    "1.1.15",
-    "1.1.16",
-]
-
-DOJO_VERSIONS = [
-    "1.4.0",
-    "1.4.1",
-    "1.4.3",
-    "1.5",
-    "1.5.1",
-    "1.6",
-    "1.6.1",
-    "1.7.0",
-    "1.7.1",
-    "1.7.2",
-    "1.7.3",
-    "1.8.0",
-    "1.8.1",
-    "1.8.2",
-    "1.8.3",
-    "1.8.4",
-    "1.8.5",
-    "1.8.6",
-    "1.9.0",
-    "1.9.1",
-    "1.9.2",
-    "1.9.3",
-]
+        "allowed", "PersonasInteractive_bootstrap.js")
+
+
+BACKBONE_VERSIONS = {
+    "ALLOWED" : [
```

Comment:
It might be a good idea to put these rule names in constants so that they don't get typod. We also don't normally put a space before the `:` in dicts.

```py
ALLOWED = "ALLOWED"
WARNING = "WARNING"

BACKBONE_VERSIONS = {
    ALLOWED: [
        "1.0.0",
        ...
    ]
}